src/DittoMachine.sol

```diff
@@ -294,29 +336,44 @@ contract DittoMachine is ERC721, ERC721TokenReceiver, ERC1155TokenReceiver {
             forceTransferFrom(ownerOf[cloneId], msg.sender, cloneId); // EXTERNAL CALL
         }
 
-        return cloneId;
+        return (
+            cloneId,
+            protoId,
+            index
+        );
     }
 
     /**
      * @notice unwind a position in a clone.
-     * @param _cloneId specifies the clone to burn.
+     * @param cloneId specifies the clone to be burned.
      * @dev will refund funds held in a position, subsidy will remain for sellers in the future.
      */
-    function dissolve(uint256 _cloneId) public {
-        if (!(msg.sender == ownerOf[_cloneId]
-                || msg.sender == getApproved[_cloneId]
-                || isApprovedForAll[ownerOf[_cloneId]][msg.sender])) {
+    function dissolve(uint256 cloneId, uint256 protoId, uint256 index) public {
```

Comment:
can we remove `cloneId` from the arguments since it can be derived from the other two arguments?